.github/workflows/make-release.yml

```diff
@@ -127,3 +127,80 @@ jobs:
         with:
           body_path: ${{github.workspace}}/release.md
           prerelease: ${{steps.extract_prerelease.outputs.prerelease}}
+  createhomebrew:
+    runs-on: ubuntu-latest
+    needs: [makerelease]
+    name: createrelease
+    steps:
+      - uses: actions/checkout@master
+      - name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0.8.0'
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CLOUD_DL_SA }}'
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0.6.0
+        with:
+          project_id: kittycadapi
+      - name: Install deps
+        shell: bash
+        run: |
+          ./.github/workflows/cross-deps.sh
+      - name: bundle binaries and make formula
+        shell: bash
+        run: |
+          mkdir homebrew
+          cp homebrew-template.rb homebrew/kittycad.rb
+
+          input_names=(
+            "x86_64-apple-darwin"
+            "aarch64-apple-darwin"
+            "x86_64-unknown-linux-musl"
+            "aarch64-unknown-linux-musl"
+          )
+          homebrew_names=(
+            "x86_64_darwin"
+            "aarch64_darwin"
+            "x86_64_linux"
+            "aarch64_linux"
+          )
+
+          to_zip_files=""
+          version=v$(toml get Cargo.toml package.version | jq -r .)
+          sed -i "s#replace-semver#$version#g" "./homebrew/kittycad.rb"
+
+          for i in "${!input_names[@]}"; do
+            input_name="${input_names[$i]}"
+            homebrew_name="${homebrew_names[$i]}"
+            
+            mkdir "./homebrew/$homebrew_name"
+            curl -L "https://dl.kittycad.io/releases/cli/$version/kittycad-$input_name" -o "./homebrew/$homebrew_name/kittycad"
+
+            sha256=$(sha256sum "./homebrew/$homebrew_name/kittycad")
+            hash=$(printf '%s\n' "$sha256" | cut -d' ' -f1)
+            sed -i "s#replace-$homebrew_name-sha#$hash#g" "./homebrew/kittycad.rb"
+
+            to_zip_files="$to_zip_files $homebrew_name/kittycad"
+          done
+
+          (cd ./homebrew && tar -czvf kittycad-cli.tar.gz $to_zip_files)
+
+          sha256=$(sha256sum "./homebrew/kittycad-cli.tar.gz")
+          hash=$(printf '%s\n' "$sha256" | cut -d' ' -f1)
+          sed -i "s#replace-tarball-sha#$hash#g" "./homebrew/kittycad.rb"
+
+          # clean up
+          for homebrew_name in "${homebrew_names[@]}"; do
+            rm -rf "./homebrew/$homebrew_name"
+          done
+
+          # verify output, debug code remove later. 
+          echo "tarball and formula"
+          (cd ./homebrew && ls)
+          echo "formula contents"
+          cat ./homebrew/kittycad.rb
+      - name: 'upload binary files'
```

Comment:
One thing I'm a little confused about is how the semver gets into the url with upload?